app/Http/Controllers/PhotoController.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\repositories\PhotoAlbumRepository;
+use App\repositories\RepositorieFactory;
+
+class PhotoController extends Controller
+{
+    private $_PhotoAlbumRepository;
+    private $_PhotoRepository;
+
+    public function __construct(RepositorieFactory $repositorieFactory){
+        $this->_PhotoAlbumRepository = $repositorieFactory->getRepositorie(RepositorieFactory::$PHOTOALBUMREPOKEY);
+        $this->_PhotoRepository = $repositorieFactory->getRepositorie(RepositorieFactory::$PHOTOREPOKEY);
+        $this->middleware('auth');
+    }
+
+    public function index($id)
+    {
+        PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->find($id);
+        $photoAlbum->description = str_replace("\r\n","<br>", $photoAlbum->description);
+        $photos = $this->getPhotos($id); 
+        $curPageName = $photoAlbum->title;
+        return view('front-end.photo_album', compact('curPageName', 'photoAlbum', 'photos'));
+    }
+
+    //List of photos.
+    public function getPhotos($id){
+        PhotoAlbum:$photoAlbum = $this->_PhotoAlbumRepository->find($id);
+        if($photoAlbum != null){
+            $photosObjects = $this->_PhotoAlbumRepository->getThumbnails($photoAlbum->id);
+            return $photosObjects;
+        } else{
+            abort(404);
+        }
+    }
+
+    public function addAlbum(Request $request){
```

Comment:
Ik zou de title en description lengte checken voordat de row in de database aangemaakt wordt, zodat we een duidelijke error message terug kunnen geven ipv een sql error.

Reply:
fixed in commit 4eca72f